### PR TITLE
Topology Manager docs update for 1.19

### PR DESCRIPTION
**Pull request by klueska**
Signed-off-by: Kevin Klues <kklues@nvidia.com>

### CODE REVIEW

**Comment by annajung** [comment]
Hi @klueska, docs shadow here 👋 just a friendly reminder this PR needs tech and doc review by July 16th.

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 1]
```diff
@@ -103,9 +106,28 @@ The general workflow of a device plugin includes the following steps:
...
+        // PreStartContainer is called, if indicated by Device Plugin during registeration phase,
+        // before each container start. Device plugin can run device specific operations
+        // such as resetting the device before making devices available to the container.
+        rpc PreStartContainer(PreStartContainerRequest) returns (PreStartContainerResponse) {}
   }
   ```
 
+  **Note:** Plugins are not required to provide useful implementations for
+  `GetPreferredAllocation()` or `PreStartContainer()`. Flags indicating which
+  (if any) of these calls are available should set in the `DevicePluginOptions`
```
Comment (nolancon) [inline content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:126]: nit: /should set/should be set/

Comment (klueska) [reply content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:126]: Updated

**Comment by klueska** [comment]
@annajung, @nolancon has provided the necessary tech review and added his LGTM. Who do I need to ping for the doc review?

**Comment by annajung** [comment]
Hi @klueska, you can reach out to docs team at #sig-docs channel to request a review, please let me know if you need any help

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md [hunk 2]
```diff
@@ -103,9 +106,28 @@ The general workflow of a device plugin includes the following steps:
...
+        rpc PreStartContainer(PreStartContainerRequest) returns (PreStartContainerResponse) {}
   }
   ```
 
+  **Note:** Plugins are not required to provide useful implementations for
+  `GetPreferredAllocation()` or `PreStartContainer()`. Flags indicating which
+  (if any) of these calls are available should be set in the `DevicePluginOptions`
+  message sent back by a call to `GetDevicePluginOptions()`. The `kubelet` will
+  always call `GetDevicePluginOptions()` to see which optional functions are
+  available, before calling any of them directly.
```
Comment (sftim) [inline content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md:129]:
```suggestion
{{% note %}}
Plugins are not required to provide useful implementations for
`GetPreferredAllocation()` or `PreStartContainer()`. Flags indicating which
(if any) of these calls are available should be set in the `DevicePluginOptions`
message sent back by a call to `GetDevicePluginOptions()`. The kubelet
always calls `GetDevicePluginOptions()` to see which optional functions are
available, before calling any of them directly.
{{% /note %}}
```

Use a [Note](https://kubernetes.io/docs/contribute/style/style-guide/#note) shortcode (untested).

Comment (klueska) [reply content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md]: 1. Updated per your suggestion with 1 minor difference. I used `{{< note >}}` not `{{% note %}}` as the link you provided suggested.
2. No, this actually moved to beta in 1.18, but the documentation was never updated to reflect that. The feature gate was toggled to "always-on" back before 1.18 was released.

Comment (sftim) [reply content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md]: On detail 1, I did also file https://github.com/kubernetes/website/issues/22488 about the advice being not quite right for callout shortcodes. Anyway, it renders well enough as is.

Comment (klueska) [reply content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md]: It looks good to me: https://deploy-preview-21607--kubernetes-io-vnext-staging.netlify.app/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/

**Comment by klueska** [comment]
Hello @sig-docs-en-reviews I am looking for an approval on this PR. It's already got a LGTM from a technical perspective.

**Comment by klueska** [comment]
Hello @rajakavitha1, @jimangel, @stewart-yu, @xiangpengzhao, @zhangxiaoyu-zidif

I am looking for an approval on this PR. It's already got a LGTM from a technical perspective.

**Comment by klueska** [comment]
Hello, @kubernetes/sig-docs-en-owners 

I am looking for an approval on this PR. It's already got a LGTM from a technical perspective.

**Pull request merged**